test/validation/api/timer/timer.c

```diff
@@ -1411,23 +1510,30 @@ static void timer_test_all(void)
 	uint64_t ns, tick, ns2;
 	uint64_t res_ns, min_tmo, max_tmo;
 	uint32_t timers_allocated;
-	pthrd_arg thrdarg;
+	struct thread_args thr_args;
 	odp_pool_capability_t pool_capa;
 	odp_timer_capability_t timer_capa;
+	odp_schedule_capability_t sched_capa;
 	odp_pool_t tbp;
 	odp_timer_pool_t tp;
 	uint32_t num_timers;
+	uint32_t num_workers;
 	int timers_per_thread;
 
+	CU_ASSERT_FATAL(odp_schedule_capability(&sched_capa) == 0);
 	/* Reserve at least one core for running other processes so the timer
 	 * test hopefully can run undisturbed and thus get better timing
 	 * results. */
-	int num_workers = odp_cpumask_default_worker(&unused, 0);
+	num_workers = odp_cpumask_default_worker(&unused, 0);
 
 	/* force to max CPU count */
 	if (num_workers > MAX_WORKERS)
 		num_workers = MAX_WORKERS;
 
+	if (queue_type == ODP_QUEUE_TYPE_SCHED &&
+	    num_workers > sched_capa.max_groups)
+		num_workers = sched_capa.max_groups;
```

Comment:
Scheduler API has 3 automatic groups (ODP_SCHED_GROUP_ALL, etc). So, application may be able to create only max_groups - 3 groups.